guide-gen/guide-configs/app-dev/observability.json

```diff
@@ -2,9 +2,10 @@
   "file_name": "observability.md",
   "id": "observability",
   "file_dir": "application-development",
-  "title": "Application development observability",
+  "title": "Application development - Observability",
   "sidebar_label": "Observability",
-  "description": "This section provides documentation on how to observe your Temporal application.",
+  "description": "The Observability section of the Temporal Application development guide covers the many ways to view the current state of your Temporal Application – that is, ways to view what Workflow Executions are tracked by the Platform and the state of any given Workflow Execution, either currently or at points of an execution.",
```

Comment:
In features.json and foundations.json, the section name is lowercase; here, Observability is capitalized. For consistency, let's choose one style.

Related: Adding a hyphen to the title in worker-performance.md.

Reply:
Sounds like we are converging on capitalizing these high-level guide section names